### PR TITLE
Update Gigalixir deployment guide

### PR DESCRIPTION
**Pull request by filiptelak**
Fixes https://github.com/phoenixframework/phoenix/issues/4474

### CODE REVIEW

File: guides/deployment/gigalixir.md [hunk 1]
```diff
@@ -96,7 +96,19 @@ $ echo 'node_version=12.16.3' > phoenix_static_buildpack.config
 If you are using Phoenix v1.6, there is also a `mix assets.deploy` alias in your `mix.exs` that you want to invoke post compilation:
 
 ```console
-$ echo 'hook_post_compile="mix assets.deploy && rm -f _build/esbuild"' >> elixir_buildpack.config
+$ echo 'hook_post_compile="mix assets.deploy && rm -f _build/esbuild"' >> phoenix_static_buildpack.config
```
Comment (josevalim) [inline guides/deployment/gigalixir.md:99]: I think we simply don't want to do this line at all. :)

Comment (jesseshieh) [reply guides/deployment/gigalixir.md:99]: Agreed =)

File: guides/deployment/gigalixir.md [hunk 2]
```diff
@@ -96,7 +96,19 @@ $ echo 'node_version=12.16.3' > phoenix_static_buildpack.config
...
+```
+
+Additionally, if you have NPM packages installed using an `assets/package.json` file, you must add a `deploy` script to it:
```
Comment (jesseshieh) [inline guides/deployment/gigalixir.md:102]: I think, at least for now, this "empty" `assets/package.json` file should be created regardless of if you have additional NPM packages installed or not. This is to make sure `mix assets.deploy` is run.

File: guides/deployment/gigalixir.md [hunk 3]
```diff
@@ -96,7 +96,19 @@ $ echo 'node_version=12.16.3' > phoenix_static_buildpack.config
...
+
+```json
+{
+  ...
```
Comment (jesseshieh) [inline guides/deployment/gigalixir.md:106]: I would remove the two sets of ellipsis here to make this snippet fully usable as-is.

**Review by jesseshieh — CHANGES_REQUESTED** [review]

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by jesseshieh** [comment]
Thanks @josevalim!

**Comment by filiptelak** [comment]
Thanks @josevalim and @jesseshieh!

**Pull request merged**